arbnode/node.go

```diff
@@ -339,6 +339,29 @@ func checkArbDbSchemaVersion(arbDb ethdb.Database) error {
 	return nil
 }
 
+func DataposterOnlyUsedToCreateValidatorWalletContract(
+	ctx context.Context,
+	l1Reader *headerreader.HeaderReader,
+	transactOpts *bind.TransactOpts,
+	cfg *dataposter.DataPosterConfig,
+	parentChainID *big.Int,
+) (*dataposter.DataPoster, error) {
+	return dataposter.NewDataPoster(ctx,
+		&dataposter.DataPosterOpts{
+			HeaderReader: l1Reader,
+			Auth:         transactOpts,
+			Config: func() *dataposter.DataPosterConfig {
+				cfg.UseNoOpStorage = true
```

Comment:
Let's mutate this outside of the closure at the top of the function, and/or make a copy of the config and mutate that copy. As-is we're modifying the config passed in potentially at random points in the future from a separate goroutine which seems like it might cause bugs, even though the callsite is pretty simple.

Reply:
Done.